### ISSUE TITLE
TinyMCE nested lists produce invalid HTML (ul child of ul)

### ISSUE THREAD

**Issue by jdufresne** [tinymce/tinymce#3076]
Steps to reproduce:
- Start with the fiddle https://jsfiddle.net/cayy8tvk/
- Click "Bullet list" toolbar icon
- Enter "foo", hit enter
- Click "Increase indent" toolbar icon _twice_
- Using the browser's Inspector, inspect the last bullet point

The HTML will look something like:

```
<ul><li>foo<br data-mce-bogus="1"></li><ul><ul><li><br data-mce-bogus="1"></li></ul></ul></ul>
```

Notice the `<ul><ul>`. This is invalid HTML. You can verify this by using the [W3C HTML5 validator](https://validator.w3.org/nu/) which reports:

> Error: Element ul not allowed as child of element ul in this context.

Additionally, this can be verified by the [HTML5 spec](https://www.w3.org/TR/html5/grouping-content.html#the-ul-element) which lists `ul`'s possible children as `li` only.

---

I think I understand why TinyMCE wants to do this, to avoid double bullet points with an additional `li` tag.

The issue I'm hitting, all content created by TinyMCE client side is validated and cleansed server side. One aspect of this cleansing is to transform any invalid HTML to valid HTML. The tag sequence `<ul><ul><li>` is converted to the valid `<ul><li><ul><li>` which, when rendered as plain HTML, has an extra bullet point.

I think an alternative solution to injecting invalid HTML would be to add `style="list-style-type:none` to empty `<li>` tag. What do you think?

**Comment by jayarjo** [comment]
@jdufresne can you please describe scenario in which you need deep (double, triple) indentation for list item?

**Comment by jdufresne** [comment]
@jayarjo I don't personally do this, but I noticed users of my application doing this. I'm not sure the specific reason they choose to format the text this way. Perhaps they like the look better.

**Comment by MichaelFromin** [comment]
I believe that if you load the *lists* plugin this will cease to happen.

**Comment by valorize** [comment]
Would be great if that would be the default. Or are there any downsides of using the lists plugin?

**Comment by dieper** [comment]
This has been getting me crazy, and I am not sure about what to do.
I have a list in the editor like this:
![screen shot 2017-10-30 at 23 53 17](https://user-images.githubusercontent.com/8155694/32199595-8ba2f9cc-bdcd-11e7-8f65-46b6acc486f8.png)
The list html is not correct and when I process it on server side using the DOMDocument object and I try to get the content of the "ol" element with `$node->ownerDocument->saveHTML($node)`.

After doing this, I got the whole html as
```
<ol>
<li>ordered list item one</li>
</ol>
<ul>
<li>indented list item one</li>
</ul>
```

The DOMDocument insert an `</ol>` ending tag before the indented list because it understand as a malformed html, returning two separated lists. So, as I see it, DOM recognises that there is some tag missing.

I need to let users make 2 or 3 level lists but now it is impossible with correct final html.

(I am using 4.2.8 version, but after reading the "changelog", I didn't notice this issue fixed).

Any info that helps a bit, will be really appreciated.

**Comment by patricknelson** [comment]
Thanks @MichaelFromin, your fix enabling the `lists` plugin works for me. For the record, I'm on an ancient version of TinyMCE `3.5.11`.

**Comment by dlepold** [comment]
i am trying tinymce on the cloud version of and i got the lists plugin installed, and it's a mess, it produces endless <ul><ul><ul> , anytime i save it converts the content and adds even more ul-tags etc. geez. 
So anyone got a sulution. Sad day ;( I hope this might be fine, sinc ethy tak money for it? :(

**Comment by owaisahmed5300** [comment]
@dlepold check this link
https://community.tiny.cloud/communityQuestion?id=90661000000IecDAAS

**Comment by lnewson** [comment]
Hi,

There was a bit of a rewrite of the lists plugin in TinyMCE 4.9.0 and I'm no longer able to reproduce the issue originally reported using TinyMCE 4.9.11 or 5.4.1. As such it's likely this was either fixed by 4.9.0 or fixed in another release, so I'm going to close this. If you're still able to reproduce the original issue though, then let us know and we'll reopen this issue.

Thanks!
